### ISSUE TITLE
bug AvalancheDataset with multi-task labels

### ISSUE THREAD

**Issue by AntonioCarta** [ContinualAI/avalanche#654]
```
import torch
from avalanche.benchmarks.utils import AvalancheTensorDataset
x = torch.rand(32, 10)
y = torch.rand(32, 10)
t = torch.ones(32)  # single task
data = AvalancheTensorDataset(x, y, targets=1, task_labels=t)
print(len(data.task_set))
```
expected: 1
result: 32

**Comment by vlomonaco** [comment]
@AndreaCossu can you add your failing example as well here?

**Comment by lrzpellegrini** [comment]
Working on it!

**Comment by AndreaCossu** [comment]
I'll open a new issue since my error does not seem to be strictly related to `AvalancheDataset`

**Comment by lrzpellegrini** [comment]
Ok, I fixed the error. It's related to the fact that task labels weren't converted to ints (AvalancheDataset kept them a single element float tensors). I'm opening a PR for this.
